services-core/src/contracts/stablex_contract.rs

```diff
@@ -147,14 +143,14 @@ pub trait StableXContract: Send + Sync {
         block_number: Option<BlockNumber>,
     ) -> Result<U256>;
 
-    fn submit_solution<'a>(
-        &'a self,
+    async fn submit_solution(
```

Comment:
These changes were because of the following lints:
```
warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
   --> services-core/src/contracts/stablex_contract.rs:150:5
    |
150 | /     fn submit_solution<'a>(
151 | |         &'a self,
152 | |         batch_index: u32,
153 | |         solution: Solution,
...   |
156 | |         nonce: U256,
157 | |     ) -> BoxFuture<'a, Result<(), MethodError>>;
    | |_______________________________________________^
    |
    = note: `#[warn(clippy::needless_lifetimes)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
```

Instead of removing the explicit lifetimes, I decided to make it use `async-trait` as it also would make the problem go away and touch an equivalent amount of code.

Reply:
The reason this was BoxFuture instead of async fn is that we couldn't test some interactions with mockall otherwise. For example with an async fn mockall didn't let us return future::pending. If the tests still pass then async fn is better.